samples/data-sealing/common/keys.cpp

```diff
@@ -96,54 +96,15 @@ oe_result_t ecall_dispatcher::get_seal_key_by_policy(
     size_t* key_info_size)
 {
     oe_result_t result = OE_OK;
-    uint8_t temp_buf[1];
     uint8_t* buf = NULL;
     size_t buf_size = 0;
     uint8_t* info = NULL;
     size_t info_size = 0;
 
     TRACE_ENCLAVE("get_seal_key_by_policy: %d", policy);
 
-    // query key buffer size and allocate memory for key buffer
     result = oe_get_seal_key_by_policy(
-        (oe_seal_policy_t)policy, temp_buf, &buf_size, NULL, NULL);
-    if (result != OE_BUFFER_TOO_SMALL)
-    {
-        TRACE_ENCLAVE(
-            "oe_get_seal_key_by_policy failed with %s\n",
-            oe_result_str(result));
-        goto exit;
-    }
-
-    buf = (uint8_t*)malloc(buf_size);
-    if (buf == NULL)
-    {
-        result = OE_OUT_OF_MEMORY;
-        TRACE_ENCLAVE("OE_OUT_OF_MEMORY");
-        goto exit;
-    }
-
-    // query key info size and allocate memory for key info buffer
-    result = oe_get_seal_key_by_policy(
-        (oe_seal_policy_t)policy, buf, &buf_size, temp_buf, &info_size);
-    if (result != OE_BUFFER_TOO_SMALL)
-    {
-        TRACE_ENCLAVE(
-            "oe_get_seal_key_by_policy failed with %s\n",
-            oe_result_str(result));
-        goto exit;
-    }
-    info = (uint8_t*)malloc(info_size);
-    if (info == NULL)
-    {
-        result = OE_OUT_OF_MEMORY;
-        TRACE_ENCLAVE("OE_OUT_OF_MEMORY");
-        goto exit;
-    }
-
-    // Query required key buffer size by passing 0 in key_buf_size
-    result = oe_get_seal_key_by_policy(
-        (oe_seal_policy_t)policy, buf, &buf_size, info, &info_size);
+        (oe_seal_policy_t)policy, &buf, &buf_size, &info, &info_size);
```

Comment:
Shouldn't we change `policy` from `int` to `oe_seal_policy_t` above instead?

Reply:
We could, the only problem it goes really deep up the stack. There are 5/6 functions which need to be changed to accommodate for it.
Please let me know if you want me to change them.

Reply:
Ah, nevermind then. Thank you!